ts/features/services/details/store/reducers/index.ts

```diff
@@ -185,48 +204,57 @@ export const serviceMetadataInfoSelector = createSelector(
     )
 );
 
-export const servicePreferencePotSelector = (state: GlobalState) =>
-  state.features.services.details.servicePreference;
-
-export const servicePreferenceResponseSuccessSelector = createSelector(
-  servicePreferencePotSelector,
-  servicePreferencePot =>
+export const servicePreferencePotByIdSelector =
+  (state: GlobalState) => (id: ServiceId | undefined) => {
+    if (id === undefined) {
+      return pot.none;
+    }
+    return state.features.services.details.preferencesById[id] ?? pot.none;
+  };
+export const servicePreferenceResponseSuccessByIdSelector = createSelector(
+  servicePreferencePotByIdSelector,
+  getServicePreferencePot => (id: ServiceId | undefined) =>
     pipe(
-      servicePreferencePot,
+      getServicePreferencePot(id),
       pot.toOption,
       O.filter(isServicePreferenceResponseSuccess),
       O.toUndefined
     )
 );
 
-export const isLoadingServicePreferenceSelector = (state: GlobalState) =>
-  pipe(
-    state,
-    servicePreferencePotSelector,
-    servicePreferencePot =>
-      pot.isLoading(servicePreferencePot) ||
-      pot.isUpdating(servicePreferencePot)
-  );
-
-export const isErrorServicePreferenceSelector = (state: GlobalState) =>
-  pipe(
-    state,
-    servicePreferencePotSelector,
-    servicePreferencePot =>
-      pot.isError(servicePreferencePot) ||
-      (isStrictSome(servicePreferencePot) &&
-        !isServicePreferenceResponseSuccess(servicePreferencePot.value))
-  );
+export const isLoadingServicePreferenceSelector = createSelector(
```

Comment:
This selector returns a boolean, there's no need to cache it using `createSelector`

Reply:
Also, let's unite the parameters together, as above